src/bun.js/webcore/blob.zig

```diff
@@ -4488,27 +4482,45 @@ pub const Blob = struct {
             // we don't need to clone
             .clone => {
                 this.store.?.ref();
+
+                if (bom == .utf16_le) {
```

Comment:
memory leak right here

Reply:
move this above this.store.?.ref() to fix the leak

Reply:
Learning moment for me. How is this a leak and why does moving it above fix it?